src/pc/common/BitArray.js

```diff
@@ -49,28 +42,13 @@ class BitArray {
   }
 
   static fromArray (array, bitsPerValue) {
-    const data = []
-    let i = 0
-    let curLong = 0
-    let curBit = 0
-    while (i < array.length) {
-      curLong |= array[i] << curBit
-      curBit += bitsPerValue
-      if (curBit > 32) {
-        data.push(curLong & 0xffffffff)
-        curBit -= 32
-        curLong = array[i] >>> (bitsPerValue - curBit)
-      }
-      i++
-    }
-    if (curBit > 0) {
-      data.push(curLong)
+    const bitarray = new BitArray({
+      capacity: array.length,
+      bitsPerValue
+    })
+    for (let i = 0; i < array.length; i++) {
+      bitarray.set(i, array[i])
```

Comment:
You are missing the point of the fromArray function, the only reason it exist is to avoid looping and calling bitarray.set().
By inlining the set function, a few redundant masking operations can be removed, yielding a more efficient implementation.

Reply:
Okay, then I don't get why BitArrayNoSpan::fromArray is implemented like that.

I'll revert that, but I still want to know

Reply:
The reason is BitArrayNoSpan::fromArray was not done yet, it should be modified the same way.